### ISSUE TITLE
[Bug]: `js_run_devserver` does not execute with updated `data` values starting with Bazel 8.4.0

### ISSUE THREAD

**Issue by scottpledger** [aspect-build/rules_js#2379]
### What happened?

Changes to the `data` attribute of `js_run_devserver` can result in I/O errors like:

```
[Error: ENOENT: no such file or directory, lstat '/private/var/tmp/_bazel_scott.pledger/90d2640d2b39ca9b655f207fb9f07c7b/execroot/_main/bazel-out/darwin_arm64-fastbuild/bin/apps/testApp/dev_/dev.runfiles/_main/apps/testApp/index.html'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/private/var/tmp/_bazel_scott.pledger/90d2640d2b39ca9b655f207fb9f07c7b/execroot/_main/bazel-out/darwin_arm64-fastbuild/bin/apps/testApp/dev_/dev.runfiles/_main/apps/testApp/index.html'
}
Trace
    at syncRecursive (file:///private/var/tmp/_bazel_scott.pledger/90d2640d2b39ca9b655f207fb9f07c7b/execroot/_main/bazel-out/darwin_arm64-fastbuild/bin/apps/testApp/dev_/dev.runfiles/aspect_rules_js+/js/private/js_run_devserver.mjs:284:17)
```

Specifically, this error is triggered by having a `js_run_devserver` rule where I run it with an `index.html` `data` entry first, then remove that entry and run it again.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: N/A

By bisecting https://github.com/bazelbuild/bazel/compare/8.3.1...8.4.0, I've narrowed this down to one of two Bazel commits:

- [8.4.0] Set working directory in interactive --sandbox_debug command - https://github.com/bazelbuild/bazel/commit/b8abd841730e7aee9abe8fb0da614315f8e5d485
- [8.4.0] Fix stale action key with expand_directories=False and manual expansion https://github.com/bazelbuild/bazel/commit/84cd58ca51fd32bb9f18433fee870f6cfad9f807

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file: 2.6.0

Language(s) and/or frameworks involved:
JS + Vite, though this seems to be an issue independent of the underlying binary and has something to do with the js_run_devserver.mjs binary itself and/or the rule that manages it.

### How to reproduce

```shell
See https://github.com/scottpledger/edit_rule_contents_repro.
```

**Comment by fmeum** [comment]
Pretty sure it's the second commit as it affects this set of args: https://github.com/aspect-build/rules_js/blob/85eedb87347757f457764e2dc8db8bd131659355/js/private/js_run_devserver.bzl#L75-L80

**Comment by fmeum** [comment]
I'll send a fix.

**Comment by fmeum** [comment]
https://github.com/bazelbuild/bazel/pull/27163 avoids this situation, but I think that there must still be a bug in `js_run_devserver`: the action may rerun for unrelated reasons such as a `bazel shutdown`.

**Comment by jbedard** [comment]
I'm going to close this since it seems to be fixed in bazel and only an issue in 8.4.

@scottpledger please comment here if you run into the issue again though.
